### ISSUE TITLE
Neuron Device Plugin nodeAffinity uses deprecated node label `beta.kubernetes.io/instance-type`

### ISSUE THREAD

**Issue by aaroniscode** [aws-neuron/aws-neuron-sdk#760]
When installing the Neuron Device Plugin on an EKS v1.28 cluster:

```
kubectl apply -f https://raw.githubusercontent.com/aws-neuron/aws-neuron-sdk/master/src/k8/k8s-neuron-device-plugin.yml
Warning: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key: beta.kubernetes.io/instance-type is deprecated since v1.17; use "node.kubernetes.io/instance-type" instead
Warning: spec.template.metadata.annotations[scheduler.alpha.kubernetes.io/critical-pod]: non-functional in v1.16+; use the "priorityClassName" field instead
daemonset.apps/neuron-device-plugin-daemonset created
```

Looks like the 2nd warning is already covered in Issue #719 and will be resolved with PR #759

For the 1st warning:

```
Warning: spec.template.spec.affinity.nodeAffinity.requiredDuringSchedulingIgnoredDuringExecution.nodeSelectorTerms[0].matchExpressions[0].key: beta.kubernetes.io/instance-type is deprecated since v1.17; use "node.kubernetes.io/instance-type" instead
```

The [Kubernetes docs](https://kubernetes.io/docs/reference/labels-annotations-taints/#beta-kubernetes-io-instance-type-deprecated) say 

> Note: Starting in v1.17, this label is deprecated in favor of [node.kubernetes.io/instance-type](https://kubernetes.io/docs/reference/labels-annotations-taints/#nodekubernetesioinstance-type).

The manifest can be updated already has a `matchExpressions` with the right label so the deprecated label can be removed.  Kubernetes 1.16 has been out of support for over 3 years (04 Aug 2020).

**Comment by aws-taylor** [comment]
Thanks @aaroniscode,

 I've forwarded this request to the appropriate team; we'll take a look.
 
 -Taylor

**Comment by micwade-aws** [comment]
Merged the change from #764, which addressed the issue.  Thanks again for highlighting this to us.
